java/daikon/dcomp/DCInstrument.java

```diff
@@ -1953,132 +2167,115 @@ InstructionList handleInvoke(InvokeInstruction invoke) {
+          // its superclass and try again.
+          mainloop:
+          while (true) {
+            // Check that the class exists
+            JavaClass targetClass;
+            try {
+              targetClass = getJavaClass(targetClassname);
+            } catch (Throwable e) {
+              targetClass = null;
+            }
+            if (targetClass == null) {
+              // We cannot locate or read the .class file, better assume not instrumented.
+              if (debugHandleInvoke) {
+                System.out.printf("Unable to locate class: %s%n%n", targetClassname);
+              }
+              callee_instrumented = false;
+              break;
+            }
+            if (debugHandleInvoke) {
+              System.out.println("target class: " + targetClassname);
+            }
+
+            for (Method m : targetClass.getMethods()) {
+              if (debugHandleInvoke) {
+                System.out.println("  " + m.getName() + Arrays.toString(m.getArgumentTypes()));
+              }
+              if (m.getName().equals(method_name)
+                  && Arrays.equals(m.getArgumentTypes(), arg_types)) {
+                // We have a match.
+                if (debugHandleInvoke) {
+                  System.out.printf("we have a match%n%n");
+                }
+                if (BcelUtil.inJdk(targetClassname)) {
+                  callee_instrumented = false;
+                }
+                break mainloop;
+              }
+            }
+
+            // no methods match - search this class's interfaces
+            @ClassGetName String found;
+            try {
+              found = getDefiningInterface(targetClass, method_name, arg_types);
+            } catch (Throwable e) {
+              // We cannot locate or read the .class file, better assume it is not instrumented.
+              callee_instrumented = false;
+              break;
+            }
+            if (found != null) {
+              // We have a match.
+              if (debugHandleInvoke) {
+                System.out.printf("we have a match%n%n");
+              }
+              if (BcelUtil.inJdk(found)) {
+                callee_instrumented = false;
+              }
+              break;
+            }
+
+            // Method not found; perhaps inherited from superclass.
+            // Cannot use "targetClass = targetClass.getSuperClass()" because the superclass might
+            // not have been loaded into BCEL yet.
+            if (targetClass.getSuperclassNameIndex() == 0) {
+              // The target class is Object; the search completed without finding a matching method.
+              if (debugHandleInvoke) {
+                System.out.printf("Unable to locate method: %s%n%n", method_name);
+              }
+              callee_instrumented = false;
+              break;
+            }
+            // Recurse looking in the superclass.
+            @SuppressWarnings("signature:assignment")
```

Comment:
Here is another place that annotations in BCEL could eliminate the warning (or where the annotations in this code should be adjusted, I'm not sure which).

Reply:
I will make this a separate pull request.